src/kademlia/routing_table.cpp

```diff
@@ -558,6 +580,10 @@ routing_table::add_node_status_t routing_table::add_node_impl(node_entry e)
 	INVARIANT_CHECK;
 #endif
 
+	// don't add if the address isn't the right type
+	if (!native_address(e.addr()))
```

Comment:
I would imagine this is a violation of a precondition of this function. In which case I think you should have an assert too